### ISSUE TITLE
Clearing fonts input data and tex data now triggers IsBuilt() assertion

### ISSUE THREAD

**Issue by Flix01** [ocornut/imgui#4455]
**Version/Branch of Dear ImGui:**

Version: 1.84.1
Branch: master

**Back-end/Renderer/Compiler/OS**

Back-ends: custom
Compiler: g++
Operating System: Ubuntu 20.04

**Issue**

Not a big deal to be honest. 
While updating my code to the current version, I've just discovered that the two lines of code I used to reduce font memory can't be used anymore, because they trigger an assertion.

```c++
    ImGuiIO& io = ImGui::GetIO();
    // Here I load the font (not shown)
    IM_ASSERT(io.Fonts->Fonts.Size>0);
    IM_ASSERT(io.Fonts->TexReady);
    IM_ASSERT(io.Fonts->IsBuilt());
    // OK here

#   ifndef IMGUIBINDINGS_DONT_CLEAR_INPUT_DATA_SOON
    // Cleanup (don't clear the input data if you want to append new fonts later)    
    io.Fonts->ClearInputData();
    io.Fonts->ClearTexData();
#   endif //IMGUIBINDINGS_DONT_CLEAR_INPUT_DATA_SOON

   IM_ASSERT(io.Fonts->IsBuilt());  // triggers here
```
The solution is just to hard-code the IMGUIBINDINGS_DONT_CLEAR_INPUT_DATA_SOON definition, unless it's just an 
assertion problem.

Is that memory saving feature still supported?

**Comment by ocornut** [comment]
I believe ClearInputData() shouldn't clear TexReady...

**Comment by ocornut** [comment]
Linking to #3487 we should add tests for those two cases.

**Comment by Flix01** [comment]
OK, good to know it's still supported. In the meantime, I can just manually set TexReady to true.
Thanks for your quick feedback. Feel free to close it whenever it's the case.

**Comment by ocornut** [comment]
Pushed a fix for this 4a7c21d3302fadce991842e79eafa99fd8690e12
